### ISSUE TITLE
Downloading files slow

### ISSUE THREAD

**Issue by mitchelltech5** [667bdrm/sofiactl#18]
currently my command im using is:

/home/pi/sofiactl/sofiactl.pl  -u <user> -p <pass>-host <ip> -port <port> -c Download -bt 2022-05-27T08:45:00Z -et  2022-05-27T09:00:00Z -ch 7 --dl

I'm doing the download from the same network as the DVR is host on.   The download seems to take forever, 15 min 200MB file, takes 30 min to download....why so long??

**Comment by mitchelltech5** [comment]
@ #2 Related?

**Comment by madmanxxx** [comment]
In my case (some cheap HiSilicon-based NVR) slow downloads are not `sofiactl`'s fault - the original ActiveX is also slow. However the stream (1080p 25fps) is downloaded faster-than-realtime over WiFi, averaging at ca. 15 Mbps.

Please compare the download speeds with the original ActiveX component to be sure.

**Comment by mitchelltech5** [comment]
In my case the files are 1080N 15 fps.

@madmanxxx So you are saying that the ActiveX on the dvr itself could be the limiting factor?

I didn't think the dvr had activeX that was just the translation for windows based machines.

Also just to clarify this was done over Ethernet directly over a 1gbps switch

**Comment by madmanxxx** [comment]
No, I said that "original ActiveX is also slow", i.e. using Internet Explorer with ActiveX plugin results in the same slow downloads in my case. Never mentioned that ActiveX is on the DVR...

The thing I noticed now is that `sofiactl` is very CPU-intensive. While downloading, it consumes ~90% of one of the cores (i5-1240P). Maybe CPU power is a limiting factor in your case? What CPU are you using?

**Comment by mitchelltech5** [comment]
That could be, during testing I was using a raspberry pi 4.  Even so, there should be a faster way to optimize the download
